src/promptflow/promptflow/_telemetry/activity.py

```diff
@@ -136,10 +134,19 @@ def wrapper(self, *args, **kwargs):
             from promptflow._telemetry.telemetry import get_telemetry_logger
 
             logger = get_telemetry_logger()
+            # TODO(2699383): use same request id with service caller
+            # by pass request id from kwargs
+            request_id = kwargs.get(REQUEST_ID_PARAM, uuid.uuid4())
+            kwargs[REQUEST_ID_PARAM] = request_id
+            custom_dimensions["request_id"] = request_id
+
+            # check if it's first SDK call
+            inner_call = kwargs.get(INNER_CALL_PARAM, False)
```

Comment:
`_inner_call=True`,` **kwargs`,` kwargs.pop(INNER_CALL_PARAM, None)`

Are we adding these codes to a large number of functions just for use here?

Is it necessary to add these codes when adding or modifying functions in the future?

I feel that these multiple functions are a bit too coupled, and those who continue writing code may have missed adding these codes.

Reply:
Yes, according to the offline discussion before, that's the only option to distinguish user calls and system calls.

Reply:
It seems that` _inner_call=False` does not need to pass through to all downstream functions? I see that the current settings are fixed values. Can we add a parameter to the decorator?

![image](https://github.com/microsoft/promptflow/assets/23182548/f60d4168-e2c7-4bad-8887-ce0f98e552e2)

Reply:
Where do we set this to True if default to False?

Reply:
After function adding this decorator, can we use it directly like this? For example:
`self.run_operations.get(name=run.name, `_inner_call=True`)`